homeworks/szabolcsgesztesi/3_conditionals_loops/flat_finder.py

```diff
@@ -0,0 +1,22 @@
+
+good_cities = ["New York", "San Fransisco"]
+worst_city = "Washington"
+best_city = "Chicago"
+
+city = input("Kérem adja meg a várost: ")
+price = int(input("Kérem adja meg a havi lakbér árát (USD): "))
+
+if city == best_city:
+    print(f"Sarah kedvenc városa {best_city}, bármi pénzert odaköltözne.")
+elif city == worst_city:
+    print(f"Sajnos Saharah számára {worst_city} elfogadhatatlan.")
+elif city in good_cities:
+    if price < 4000:
+        print(f"{city} elfogadható Sarah számára {price} dollárért.")
+    else:
+        print(f"{city} nem elfogadható Sarah számára {price} dollárért.")
+else:
```

Comment:
A végén ezt az else + if kombót össze is olvaszthatod, logikailag helyes marad, és egyszerűsődik a kódod:

```
elif price < 3000:
    print(f"{city} elfogadható Sarah számára {price} dollárért.")
else:
    print(f"{city} nem elfogadható Sarah számára {price} dollárért.")
```

Reply:
Megírtam egy sorban az utolsót! :)

Itt egyébként a fentebb lévő "elif city in good_cities:" alatt miért nem lehet 1 sorban megírni? Megpróbáltam de nem nem lesz úgy jó, muszáj kiírni "if"-el.

Reply:
Szia! Nem arra gondoltam, hogy használj egy ternary operator-t, hanem hogy nincs szükség arra hogy így írd:

```
else:
    if:
       ....
    else:
       ....
```

Hanem itt az else és az if egybe olvasztható:

```
elif:
    ....
else:
    ....
```
 Ezt írtam felül:
```
elif price < 3000:
    print(f"{city} elfogadható Sarah számára {price} dollárért.")
else:
    print(f"{city} nem elfogadható Sarah számára {price} dollárért.")
```
 
Működik ez a megoldásod is, de a ternary operatornak pont az a lényege hogy ne kelljen egy if-elif-else condition-t leírni, hanem megoldjuk egy soron. Ha már eleve van egy if-elif-else a kódban, akkor abba nem szokás egy ternary-t rakni, mert csak bonyolódik a logika.